src/main/java/org/jitsi/videobridge/octo/OctoPacket.java

```diff
@@ -85,28 +88,23 @@ private static int getMediaTypeId(MediaType mediaType)
      * Writes an Octo header to the specified buffer at the specified offset.
      * @param buf the buffer to write to.
      * @param off the offset to write at.
-     * @param r the value of the {@code r} flag.
      * @param mediaType the media type.
-     * @param s the value of the {@code s} flag.
      * @param conferenceId the Octo ID of the conference.
      * @param endpointId the Octo ID of the endpoint.
      */
     public static void writeHeaders(
             byte[] buf, int off,
-            boolean r, MediaType mediaType, int s,
-            String conferenceId,
+            MediaType mediaType,
+            long conferenceId,
             String endpointId)
     {
-        buf[off] = 0;
-        if (r)
-        {
-            buf[off] |= 0x80;
-        }
-        buf[off] |= (getMediaTypeId(mediaType) & 0x03) << 5;
-        buf[off] |= (s & 0x03) << 3;
-
-        writeConferenceId(conferenceId, buf, off, OCTO_HEADER_LENGTH);
-        writeEndpointId(endpointId, buf, off, OCTO_HEADER_LENGTH);
+        assertMinLen(buf, off, buf.length);
+        off += writeConferenceId(conferenceId, buf, off);
+        off += writeEndpointId(endpointId, buf, off);
+        buf[off] = (byte) (getMediaTypeId(mediaType) << 6);
```

Comment:
Should we have `readMediaType` and `writeMediaType`?

Reply:
We already have `readMediaType`. The `writeXXX` helpers are private, so I didn't think necessary to add one for `mediaType` (to make the semantics of `writeMediaType` nice we would have to take care of not replacing the extra bits of that byte, and then zero it out here, which is unnecessary complexity).

Reply:
Oh, oops, I saw the read code below but didn't notice it was within `readMediaType`.